app/forms/responsiblepeople/PersonNonUKPassportFormProvider.scala

```diff
@@ -36,7 +36,9 @@ class PersonNonUKPassportFormProvider @Inject()() extends Mappings {
       booleanFieldName -> boolean(emptyError, emptyError),
       "nonUKPassportNumber" -> mandatoryIfTrue(
         booleanFieldName,
-        text(s"$emptyError.number").verifying(
+        text(s"$emptyError.number")
+          .transform[String](_.replace(" ", "").trim, x => x)
```

Comment:
There is a function that has just gone in to main that would be very handy for this:

https://github.com/hmrc/amls-frontend/blob/main/app/forms/mappings/Mappings.scala#L53

If you could rebase and use this function that would be great!